sdk/typescript/src/cryptography/mnemonics.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import bip39 from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+
+export function generateMnemonic(): string {
+  return bip39.generateMnemonic(wordlist);
+}
+
+/**
+ * Sanitize the mnemonics string provided by user
+ * @param mnemonics a 12-word string split by spaces that may contain mixed cases
+ * and extra spaces
+ * @returns a sanitized mnemonics string
+ */
+export function normalizeMnemonics(mnemonics: string): string {
+  return mnemonics
+    .trim()
+    .split(/\s+/)
+    .map((part) => part.toLowerCase())
+    .join(' ');
+}
+
+/**
+ * Parse and validate a path that is compliant to SLIP-0010 in form m/44'/784'/{account_index}'/{change_index}'/{address_index}'
+ *
+ * @param path path string (e.g. `m/44'/784'/0'/0'/0'`)
+ */
+export function isValidHardenedPath(path: string): boolean {
+  if (!new RegExp("^m\\/44'\\/784'\\/0'\\/[0-9]+'\\/[0-9]+'+$").test(path)) {
```

Comment:
Another option here would be to make a version of this regex with a group that matches either 44 or 54, then these validation functions can just use the capture group result to determine if they match. The benefit of that would be one regex to maintain vs two, although I don't imagine these changing often.

Reply:
there is a hyphen for the last two levels (m/54'/784'/0'/0/0 vs m/44'/784'/0'/0'/0') that is also different, it means it is hardened, that actually differentiate BIP32 and SLIP10. It feels cleaner to keep the validation logic separate to me.